var/spack/repos/builtin/packages/geant4/package.py

```diff
@@ -41,27 +37,19 @@ class Geant4(CMakePackage):
 
     # C++11 support
     depends_on("clhep@2.4.0.0 cxxstd=11", when="@10.04 cxxstd=11")
-    depends_on("clhep@2.3.4.3 cxxstd=11", when="@10.03.p03 cxxstd=11")
-    depends_on("clhep@2.3.1.1 cxxstd=11", when="@10.02.p01 cxxstd=11")
-    depends_on("clhep@2.3.1.1 cxxstd=11", when="@10.02.p01 cxxstd=11")
-    depends_on("clhep@2.2.0.4 cxxstd=11", when="@10.01.p03 cxxstd=11")
+    depends_on("clhep@2.3.4.6 cxxstd=11", when="@10.03.p03 cxxstd=11")
     depends_on("vecgeom cxxstd=11", when="+vecgeom cxxstd=11")
 
     # C++14 support
     depends_on("clhep@2.4.0.0 cxxstd=14", when="@10.04 cxxstd=14")
-    depends_on("clhep@2.3.4.3 cxxstd=14", when="@10.03.p03 cxxstd=14")
-    depends_on("clhep@2.3.1.1 cxxstd=14", when="@10.02.p02 cxxstd=14")
-    depends_on("clhep@2.3.1.1 cxxstd=14", when="@10.02.p01 cxxstd=14")
-    depends_on("clhep@2.2.0.4 cxxstd=14", when="@10.01.p03 cxxstd=14")
+    depends_on("clhep@2.3.4.6 cxxstd=14", when="@10.03.p03 cxxstd=14")
     depends_on("vecgeom cxxstd=14", when="+vecgeom cxxstd=14")
 
     # C++17 support
-    patch('cxx17.patch', when='cxxstd=17')
+    patch('cxx17.patch', when='@:10.03.zzzz cxxstd=17')
```

Comment:
Is the addition of `zzzz` here intentional? What does it do?

Reply:
Yes. The wrong patch was being picked otherwise.

Reply:
I think the with spack version syntax I need 
when=@10.03.p99 
just to be safe.

Reply:
See my comment on the next line.

Reply:
Can you point me to the explanation of this syntax? I've never seen it before.

Reply:
I saw a discussion on the packaging slack channel using spack python to understand Version comparisons.